core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -116,6 +117,43 @@ public void calculateByRoundRobin(Map<String, String> subnetAzPairs, InstanceGro
         }
     }
 
+    public Map<String, String> filterSubnetByLeastUsedAz(InstanceGroup instanceGroup, Map<String, String> subnetAzPairs) {
+        Map<String, List<String>> azSubnetPairs = createAzSubnetPairsFromSubnetAzPairs(subnetAzPairs);
+        Map<String, Integer> azUsage = new HashMap<>();
+        Set<String> azs = azSubnetPairs.keySet();
+        initializeSubnetUsage(new HashSet<>(subnetAzPairs.values()), azs, azUsage);
+        collectCurrentAzUsage(instanceGroup, azUsage, subnetAzPairs);
+        Integer numberOfInstanceInASubnet = searchTheSmallestInstanceCountForSubnets(azUsage);
+        String leastUsedAz = searchTheSmallestUsedSubnetID(azUsage, numberOfInstanceInASubnet);
+        Set<String> subnetForAz = new HashSet<>(azSubnetPairs.get(leastUsedAz));
+        return subnetForAz.stream().collect(Collectors.toMap(subnetId -> subnetId, v -> leastUsedAz));
```

Comment:
Not a big deal but perhaps `subnetId -> subnetId` could be simplified to `Function.identity()`.